### PR TITLE
Support for link when ElementCount > 0

### PR DESCRIPTION
**Pull request by Jolbas**
When a App::Link property ElementCount was set to 1 or higher, the macro couldn't follow and the macro reported the placement of the root object instead.

### CODE REVIEW

File: Information/GetGlobalPlacement.FCMacro [hunk 1]
```diff
@@ -16,62 +16,19 @@ __Requires__ = 'FreeCAD >=0.19'
...
-    The last part of ``subobject_fullpath`` is then a specific vertex, edge or
-    face and is ignored.
-    So, subobject_fullpath has the form 'name0.name1.Edge001', for example; in
-    this case, the returned objects are
-    [object_named_name0, object_named_name1].
+def get_global_placement_and_scale(object: app.DocumentObject,
+                         subobject_fullpath: str,
+                         ) -> Tuple[app.Placement, app.Vector]:
+    """Return the global placement and the total scale, respecting links.
+    This is in contrast with``object.getGlobalPlacement()`` that returns the placement of the original
```
Comment (luzpaz) [inline Information/GetGlobalPlacement.FCMacro:30]: missing whitespace between 'with' and object.getGlobalPlacement()

File: Information/GetGlobalPlacement.FCMacro [hunk 2]
```diff
@@ -78,28 +42,38 @@ def get_subobjects_by_full_name(
...
-            return []
-        objects.append(subobject)
-    return objects
-
-
-def get_global_placement(root_object: app.DocumentObject,
+    
+    mx = None
+    if subobject_fullpath[-1] == '.':
+        # This seems to be nessecary on ``Part::*`` objects,``PartDesign::Body' etc.
```
Comment (luzpaz) [inline Information/GetGlobalPlacement.FCMacro:49]: space after comma

**Review by luzpaz — CHANGES_REQUESTED** [review]

File: Information/GetGlobalPlacement.FCMacro [hunk 3]
```diff
@@ -78,28 +42,38 @@ def get_subobjects_by_full_name(
...
-        if subobject is None:
-            # This should not append.
-            return []
-        objects.append(subobject)
-    return objects
-
-
-def get_global_placement(root_object: app.DocumentObject,
+    
+    mx = None
```
Comment (galou) [inline Information/GetGlobalPlacement.FCMacro:47]: `mx` as a variable name is not relevant. Can you be more explicit?

**Review by galou — CHANGES_REQUESTED** [review]
Thanks for the contribution.

I didn't finish my review because I noticed that your version returns an error with a `Part::Box`  and a `Part::Link` to it. When selecting the link I get the error:
```
14:19:02  Traceback (most recent call last):
  File "/home/gael/.local/share/FreeCAD/Macro/GetGlobalPlacementJolbas.FCMacro", line 142, in <module>
    get_and_print_selected_placements()
  File "/home/gael/.local/share/FreeCAD/Macro/GetGlobalPlacementJolbas.FCMacro", line 132, in get_and_print_selected_placements
    wo_subelement = strip_subelement(sub_fullpath)
<class 'NameError'>: name 'strip_subelement' is not defined
```

Can you please have a look at that? Thanks!

Do you have some files that didn't work with the original macro that we could use as test-case?

File: Information/GetGlobalPlacement.FCMacro [hunk 4]
```diff
@@ -113,13 +87,9 @@ def get_global_placement(root_object: app.DocumentObject,
...
             "additive primitve" in PartDesign.
-
     """
-    objects = get_subobjects_by_full_name(root_object, subobject_fullpath)
-    placement = root_object.Placement
-    for o in objects:
-        placement *= o.Placement
-    return placement
+
+    return get_global_placement_and_scale(object, subobject_fullpath)[0]
```
Comment (galou) [inline Information/GetGlobalPlacement.FCMacro:92]: The case where `get_global_placement_and_scale()` returns `None` is not handled.

**Comment by Jolbas** [comment]
Thanks for test and review. I have a look.

**Comment by galou** [comment]
I went ahead and integrated your changes. 
@Jolbas can you double-check my changes?
@Jolbas, I added you as author of the macro and commit author.

**Comment by galou** [comment]
@Jolbas I added a test file that helped me see what was wrong in the original macro. Maybe you can add a test with a link array.

**Comment by Jolbas** [comment]
@galou   Have done som additional fixes in https://github.com/FreeCAD/FreeCAD-macros/pull/158

**Pull request closed without merging**